tests/tt_metal/tt_metal/integration/matmul/test_matmul_X_tile.cpp

```diff
@@ -51,14 +51,14 @@ void create_test_stimuli(MatmulTileStimuli& stimuli, uint32_t M, uint32_t K, uin
     stimuli.t = tensor.get_values();
 
     auto activations_tilized = test_utils::tilize(tensor.get_values(), M * 32, K * 32);
-    auto activations_tile_layout = convert_to_tile_layout(activations_tilized);
+    auto activations_tile_layout = convert_to_tile_layout(tt::stl::MakeConstSpan(activations_tilized));
```

Comment:
Honestly, very annoyed that I have to explicitly build a const span our of a vector. 
I really wish it just worked

Reply:
https://godbolt.org/z/xMoro4f55
So, you could do this:
```suggestion
    auto activations_tile_layout = convert_to_tile_layout(tt::stl::Span{std::as_const(activations_tilized)});
```
and you wouldn't need the `std::as_const` if `activations_tilized` were `const auto`. The reason the const-qualification is necessary is because otherwise CTAD deduces `tt::stl::Span<T>` instead of `tt::stl::Span<const T>`, and then the template argument deduction on `convert_to_tile_layout` fails because even though `tt::stl::Span<T>` is implicitly convertible to `tt::stl::Span<const T>`, template argument deduction doesn't work with implicit conversion, it must match exactly.

Reply:
tl;dr blame the language, not the class template 😉

Reply:
Tbh `MakeSpan` / `MakeConstSpan` to me look much more readable than the ctor + ctad. I usually know what `T` is, but I don't like to stop and think whether ctad correctly chooses const or non const span. `MakeConstSpan` provides exactly what I need in this case.

Also this https://google.github.io/styleguide/cppguide.html#CTAD

*CTAD does give the reader more information than auto, but it also doesn't give the reader an obvious cue that information has been omitted.*

I wouldn't go as far as banning it though.

Reply:
I agree the syntactic use of CTAD is a code-smell here. Perhaps `MakeConstSpan` is the most sensical way forward, though I'd prefer the second template parameter with the default (`class T = ...`) to be refactored as a local type alias in the function template body (i.e. `using T = ...`).

Reply:
One last thing from me, not a strong preference by any means, but if we want to follow naming conventions from the standard, there are some function templates `std::as_bytes` and `std::as_writable_bytes`, perhaps we can use the names `MakeSpan` and `MakeWritableSpan` instead of `MakeConstSpan` and `MakeSpan` respectively, as I suspect the `MakeWritableSpan` will be used less often. Also it follows the de-facto principle of const-by-default.

Reply:
@patrickroberts 
>I'd prefer the second template parameter with the default (class T = ...) to be refactored as a local type alias in the function template body (i.e. using T = ...).

Makes sense, will do.

>we can use the names MakeSpan and MakeWritableSpan instead of MakeConstSpan and MakeSpan

Let me keep the MakeConstSpan for now. I am following the [absl](https://github.com/abseil/abseil-cpp/blob/master/absl/types/span.h#L726-L796) example.